### DISCUSSION TITLE
Why do the relations inside the editor sometimes break after saving?

### DISCUSSION THREAD

**Discussion by JohannesPertl** [formfcw/directus-extension-flexible-editor#40] [Q&A] [answered]
I suspect this is a bug. Often when editing a flexible editor field and saving, the relations are then broken and I have to add all of the blocks again. 
It seems to happen randomly, I couldn't reproduce it consistently.
Is this a known bug? Any way to fix or debug it?

![Untitled](https://github.com/formfcw/directus-extension-flexible-editor/assets/45853628/ba3fde47-11ea-4f6f-82be-026b1d694977)

**Comment by formfcw** [comment, marked as answer]
Since the relation block nodes in the editor reference items from a separate M2A field, there are several reasons why their reference may be lost:
- the ID of the relation block node has changed in the editor
- the ID of the referenced M2A element has changed
- the referenced M2A element has been deleted

It's not a known bug, and I'm afraid we can't fix it until we can reproduce it.

To debug: Compare the IDs for your relation block nodes (by clicking on the fields label > Edit Raw Value) with the IDs of the corresponding M2A items and make sure they match.

If you want to turn this into an issue:
- Share your Directus env vars: maybe there is something that doesn’t correlate.
- Share either your schema created by the Directus cli or a step-by-step guide on how to rebuild your collections and blocks.
